File: Python/Q46.py
# Q46.py - Program to Add Two Matrices Using Multi-dimensional Arrays
rows = int(input("Enter number of rows: "))
cols = int(input("Enter number of columns: "))
matrix1 = [list(map(int, input(f"Enter row {i+1} of first matrix: ").split())) for i in range(rows)]
matrix2 = [list(map(int, input(f"Enter row {i+1} of second matrix: ").split())) for i in range(rows)]
result = [[matrix1[i][j] + matrix2[i][j] for j in range(cols)] for i in range(rows)]
print("Sum of matrices:")
for row in result:
    print(row)

File: Java/Q46.java
/*
 Q46- Program to Add Two Matrices Using Multi-dimensional Arrays
 */

import java.util.Scanner;

public class Q46{
    public static void main(String[] args){
        Scanner sc=new Scanner(System.in);

        // Input dimensions
        System.out.print("Enter number of rows: ");
        int rows=sc.nextInt();
        System.out.print("Enter number of columns: ");
        int cols=sc.nextInt();

        int[][] a=new int[rows][cols];
        int[][] b=new int[rows][cols];
        int[][] sum=new int[rows][cols]; 

        // Input matrix A
        System.out.println("Enter elements of first matrix:");
        for(int i=0; i<rows; i++){
            for(int j=0; j<cols; j++){
                a[i][j] = sc.nextInt();
            }
        }

        // Input matrix B
        System.out.println("Enter elements of second matrix:");
        for(int i=0; i<rows; i++){
            for(int j=0; j<cols; j++){
                b[i][j] = sc.nextInt();
            }
        }

        // Add the matrices
        for(int i=0; i<rows; i++){
            for(int j=0; j<cols; j++){
                sum[i][j]=a[i][j]+b[i][j];
            }
        }

        // Display the result
        System.out.println("Sum of the two matrices:");
        for(int i=0; i<rows; i++){
            for(int j=0; j<cols; j++){
                System.out.print(sum[i][j] + " ");
            }
            System.out.println();
        }

        sc.close();
    }
}
